addons/mail/models/mail_followers.py

```diff
@@ -171,11 +171,13 @@ def _get_recipient_data(self, records, message_type, subtype_id, pids=None, cids
                 query_pid = """
 SELECT DISTINCT ON (partner.id) partner.id as pid, NULL::int AS cid,
     partner.active as active, partner.partner_share as pshare, NULL as ctype,
-    users.notification_type AS notif, NULL AS groups
+    users.notification_type AS notif, array_agg(groups.id) AS groups
```

Comment:
I think this returns {None} when having a partner without users. We could use ``ARRAY_AGG(groups_rel.gid) FILTER (WHERE groups_rel.gid IS NOT NULL) AS groups`` (returns False I think), could you check ?

Reply:
The original code returns {NULL} and your change seems to work :D

Reply:
I had checked that and left it since we do the exact same thing in the same method above, but I agree that this might be an issue (maybe this should be fixed above too), it probably never happened because it's rare in stable odoo to have a user without group

Reply:
Indeed, just that managing a False value is easier. But that's a corner case (code will be moving towards that in master). No need to change existing code I think.